scripts/west_commands/runners/dfu.py

```diff
@@ -18,12 +18,16 @@
 class DfuUtilBinaryRunner(ZephyrBinaryRunner):
     '''Runner front-end for dfu-util.'''
 
-    def __init__(self, cfg, pid, alt, img, exe='dfu-util',
+    def __init__(self, cfg, dev_id, alt, img, exe='dfu-util',
                  dfuse_config=None):
+
+        if not dev_id:
+            raise RuntimeError('Please specify a USB VID:PID with the '
+                               '-i/--dev-id command-line switch.')
```

Comment:
The error handling in the calling layers expects the constructor to succeed. Please defer raising exceptions to do_run().